cpp/sopt/wavelets/sara.cc

```diff
@@ -1,6 +1,4 @@
 #include "sara.h"
```

Comment:
All this file does is include some headers and create a namespace,there are no functions in that namespace. Why does this file exist?

Reply:
This happens in a few other files too. I think we should just merge them up (if we manage to hunt them). 

But perhaps I should do that in a separate PR (with a tag of `code=quality`)?

Reply:
@SJaffa I looked into this. It appears that this file is used by [`cpp/sopt/CMakeLists.txt` in line 26](https://github.com/astro-informatics/sopt/blob/203106724c9b05e7a5d362a21349ef1864f42301/cpp/sopt/CMakeLists.txt#L26), where it says:

```cmake
list(APPEND sources mpi/communicator.cc mpi/registered_types.cc mpi/session.cc wavelets/sara.cc)
```

I deleted this file `cpp/sopt/wavelets/sara.cc` and removed this source from the relevant line in  `cpp/sopt/CMakeLists.txt`, and it builds fine and all tests pass (locally).  I think it be better to submit the cleanup in a separate/smaller PR. Please let me know what you'd prefer.

Reply:
Thank you. Have recorded this in #393.  Should be closed by #394.